src/Avalonia.Controls/Button.cs

```diff
@@ -251,7 +251,9 @@ protected override void OnPointerReleased(PointerReleasedEventArgs e)
                 IsPressed = false;
                 e.Handled = true;
 
-                if (ClickMode == ClickMode.Release && new Rect(Bounds.Size).Contains(e.GetPosition(this)))
+                //only renderer (hittesting) know better whether pointer is over the bounds of the button
```

Comment:
Nit: I don't think this comment is really necessary - it's pretty obvious what's going on.